src/registry/routes/helpers/get-component.js

```diff
@@ -145,6 +145,15 @@ module.exports = function(conf, repository) {
           });
         }
 
+        // Support legacy templates
+        let templateType = component.oc.files.template.type;
```

Comment:
at this point, we are positive that `component.oc.files.template` is not `undefined` right?